docs/concepts/cluster_queue.md

```diff
@@ -141,9 +143,87 @@ metadata:
 ## Cohort
 
 ClusterQueues can be grouped in _cohorts_. ClusterQueues that belong to the
-same cohort can borrow unused quota from each other, if they reference
-matching resource flavors in
-`.spec.requestableResources[*].flavors[*].resourceFlavor`.
+same cohort can borrow unused quota from each other.
+
+To add a ClusterQueue to a cohort, specify the name of the cohort in the
+`.spec.cohort` field. All the ClusterQueues that have a matching `spec.cohort`
+are part of the same cohort. If the `spec.cohort` field is empty, the
+ClusterQueue doesn't belong to any cohort, and thus it cannot borrow quota from
+any other ClusterQueue.
+
+Borrowing happens per-flavor. If the ClusterQueues belonging to a cohort
+reference different flavors in `.spec.requestableResources[*].flavors[*].resourceFlavors`,
+they won't be able to share quota among them.
+
+### Borrowing semantics example
+
+Assume you created the following two ClusterQueues:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha1
+kind: ClusterQueue
+metadata:
+  name: team-a-cq
+spec:
+  namespaceSelector: {}
+  cohort: team-ab
+  requestableResources:
+  - name: "cpu"
+    flavors:
+    - resourceFlavor: default
+      quota:
+        guaranteed: 9
+  - name: "memory"
+    flavors:
+    - resourceFlavor: default
+      quota:
+        guaranteed: 36Gi
+```
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha1
+kind: ClusterQueue
+metadata:
+  name: team-b-cq
+spec:
+  namespaceSelector: {}
+  cohort: team-ab
+  requestableResources:
+  - name: "cpu"
+    flavors:
+    - resourceFlavor: default
+      quota:
+        guaranteed: 12
+  - name: "memory"
+    flavors:
+    - resourceFlavor: default
+      quota:
+        guaranteed: 48Gi
+```
+
+ClusterQueue `team-a-cq` can admit workloads depending on the following
+scenarios:
+
+- If ClusterQueue `team-b-cq` has no admitted workloads, then ClusterQueue
+  `team-a-cq` can admit workloads with resources adding up to `12+9=21` CPUs and
+  `48+36=84Gi` of memory.
+- If ClusterQueue `team-b-cq` has pending workloads and the ClusterQueue
+  `team-a-cq` has all its `guaranteed` quota used, Kueue will admit workloads in
+  ClusterQueue `team-b-cq` before admitting any new workloads in `team-a-cq`.
+  Therefore, Kueue ensures the `guaranteed` quota for `team-b-cq` is met.
```

Comment:
should we stress that there is no preemption?

Reply:
It's right in the next paragraph :)